src/Test.Utilities.Sarif/FileDiffingUnitTests.cs

```diff
@@ -256,6 +262,20 @@ private void CompareActualToExpected(
                                                      expectedSarifTextDictionary[key],
                                                      out SarifLog actual);
 
+                    if (expectedResultFingerprintKeys != null &&
+                        actual != null &&
+                        actual.Runs[0].Results != null)
```

Comment:
use this single line check:
 actual?.Runs?[0]?.Results != null

Reply:
you could also set a result local to this value.

Reply:
Updated based on recommendation